tests/test_website.py

```
@@ -31,101 +47,260 @@ def mock_config_get_side_effect(key_tuple, default=None):
-        # It should check for PDF first. Since our mock returns True for PDF,
-        # it should find it and not proceed to check for HTML.
-        calls_to_os_exists = mock_os_exists.call_args_list
+        self.mock_requests_session_get.side_effect = [
+            RequestException("Direct download failed"), 
+            mock_response_page 
+        ]
+        self.mock_beautifulsoup.return_value.select_one.return_value = None # No link found
         
-        self.assertIn(unittest.mock.call(pdf_check_path), calls_to_os_exists,
-                      "os.path.exists should have been called for the .pdf file.")
+        playwright_downloaded_path = os.path.join(MOCK_CONFIG_DATA['paths']['download_dir'], "playwright_file.pdf")
```

Ruff (0.11.9)

Local variable `playwright_downloaded_path` is assigned to but never used

Remove assignment to unused variable `playwright_downloaded_path`

(F841)